test/AWS_Tests/src/Inter_Backend_File_Operations_Spec.enso

```diff
@@ -0,0 +1,158 @@
+                Panic.with_finalizer regular_destination_file.delete_if_exists <|
+                    # The raw datalink config is copied, so reading back the .txt file yields us the configuration:
+                    regular_destination_file.read . should_contain '"libraryName": "Standard.AWS"'
+
+            group_builder.specify "should be able to copy a datalink file to another datalink" <|
+                destination_file.delete_if_exists
+                Data_Link_Format.write_raw_config source_file sample_data_link_content replace_existing=True . should_succeed
+
+                source_file.copy_to destination_file replace_existing=True . should_succeed
+                # Now the destination is _also_ a datalink, pointing to the same target as source, so reading it yields the target data:
+                destination_file.read . should_equal "Hello WORLD!"
+
+                # But if we read it raw, we can see that it is still a datalink, not just a copy of the data:
+                Data_Link_Format.read_raw_config destination_file . should_equal sample_data_link_content
+
+            group_builder.specify "should be able to move a datalink" <|
+                destination_file.delete_if_exists
+                Data_Link_Format.write_raw_config source_file sample_data_link_content replace_existing=True . should_succeed
+
+                source_file.move_to destination_file . should_succeed
+                source_file.exists . should_be_false
+
+                destination_file.read . should_equal "Hello WORLD!"
+                Data_Link_Format.read_raw_config destination_file . should_equal sample_data_link_content
+
+            group_builder.specify "should be able to move a regular file to become a datalink" <|
+                destination_file.delete_if_exists
+
+                regular_source_file = source_file.parent / source_file.name+".txt"
+                sample_data_link_content.write regular_source_file on_existing_file=Existing_File_Behavior.Overwrite . should_succeed
+                Panic.with_finalizer regular_source_file.delete_if_exists <|
+                    # The source file is not a datalink - it is a raw text file, so reading it gives us back the raw config
+                    regular_source_file.read . should_equal sample_data_link_content
+                    # Reading it raw will even fail:
+                    Test.expect_panic Illegal_Argument <|
+                        Data_Link_Format.read_raw_config regular_source_file
+
+                    regular_source_file.move_to destination_file . should_succeed
+                    regular_source_file.exists . should_be_false
+
+                    # However, the destination file _is_ a datalink, so it is read as target data:
+                    destination_file.read . should_equal "Hello WORLD!"
+                    # Unless we read it raw:
+                    Data_Link_Format.read_raw_config destination_file . should_equal sample_data_link_content
```

Comment:
Some thoughts on coyping/moving datalinks.

Currently the behaviour is to copy the raw config, not the target data. This seems to be mostly aligned with the 'datalinks as symlinks' metaphor we are using.

However, it feels like sometimes it may be useful to be able to copy actual _data_ the datalink points to.

I was thinking if we should actually differentiate the behaviour based on what the source and destination are. We can tell if a File is a datalink without reading it (and it doesn't even have to exist yet) - it's based on the file metadata, mostly just `file.extension == ".datalink"` check, alternatively content-type for HTTP and `asset_type==Data_Link` for `Enso_File` - so we could rely on that.

I'd suggest to modify the behaviour (in a followup ticket, not here) in the following way:
- if both source and destination are a data link, the data link config is copied from one place to another - _this is the currently implemented behaviour_ - sticking to the symlink metaphor.
- if both source and destination are **not** data links, the data is copied/moved as data - that is of course the current behaviour and always has been.
- _NOW the new idea is:_
	- if the source is a data link but the destination is a regular file, we'd copy the _data_ from the data link's target into the destination;
	- and if the source is a regular file and the destination is a data link, we perform a write to the data link's target of the source data (not yet possible, #9292).

Currently the behaviour is weird, as can be seen in these tests:
- if the source is a datalink but the target is not, the config is copied to the target and the target file exposes the raw datalink config;
- if the source is a regular file but the target is a datalink, that creates a new datalink that may e.g. be malformed.

I think the proposed behaviour would be better and more intuitive. What do you think? If you think this makes sense, I will create a ticket to implement it.

Reply:
This sounds good to me.

Reply:
think that makes sense

Reply:
Created a followup ticket then: #9522